pyro/contrib/named.py

```diff
@@ -207,3 +207,69 @@ def param(placeholder, *args, **kwargs):
     if not isinstance(placeholder, Object):
         return placeholder  # value was already set
     return placeholder.param(*args, **kwargs)
+
+
+@functools.wraps(pyro.condition)
+def condition(fn, data):
+    lifted_data = {}
+    def partial_condition(latent, *args, **kwargs):
+        for key, value in data.items():
+            lifted_data["{}.{}".format(latent, key)] = value
+        pyro.condition(lift(fn, _latent=latent),
+                       data=lifted_data)(*args, **kwargs)
+    return partial_condition
+
+
+def lift(fn, _namespace="latent", _latent=None):
```

Comment:
Is this an internal helper or an external function that users might want to use? If internal, please rename to `_lift`; if external, please add a docstring.